File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/usecases/likes/GetLikesPageUseCase.java
package nl.acidcats.tumblrlikes.core.usecases.likes;

import java.util.List;

import nl.acidcats.tumblrlikes.core.constants.LoadLikesMode;
import nl.acidcats.tumblrlikes.core.models.Photo;
import rx.Observable;

/**
 * Created by stephan on 13/04/2017.
 */

public interface GetLikesPageUseCase {
    Observable<List<Photo>> loadLikesPage(LoadLikesMode mode);

    Observable<Boolean> checkLoadLikesComplete();
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/core/usecases/likes/GetLikesPageUseCase.kt
package nl.acidcats.tumblrlikes.core.usecases.likes

import nl.acidcats.tumblrlikes.core.constants.LoadLikesMode
import rx.Observable

/**
 * Created on 24/10/2018.
 */
interface GetLikesPageUseCase {
    fun loadLikesPage(mode: LoadLikesMode): Observable<Long>

    fun checkLoadLikesComplete(): Observable<Boolean>
}